save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/ExecutionScoreUtils.kt

```diff
@@ -57,7 +57,6 @@ fun Int.isValidScore() = this in 0..100
 
 private fun ExecutionDto.calculateScoreForContestMode(scoreType: ScoreType): Double = when (scoreType) {
     ScoreType.F_MEASURE -> calculateFmeasure()
-    else -> TODO("Invalid score type for contest mode!")
```

Comment:
Why do you prefer such way? Of course, now exhaustive, but if we will expand our data classes, we will need to remember all such places, in aim to update them
while in other hand we will get exceptions, if we will miss something

Reply:
This way, if we miss something, we'll get a compile error (which is better than a run-time error, isn't it?)

Reply:
I expect warning in this case but

![Capture](https://user-images.githubusercontent.com/26010098/213426333-92521486-8f3c-4603-ae4e-166461a13708.JPG)

Reply:
In your case, you have an expression of type `Unit`.
A non-exhaustive `when` evaluates to `Unit`, too

In the code (`ExecutionScoreUtils.kt`), there's an expression of type `Double`

Reply:
Oh, okay, thought that we talking about general case, that each and every `when` should be exhaustive
here, I didn't pay attention into the return type